src/algebra/ordered_monoid.lean

```diff
@@ -409,6 +211,11 @@ namespace with_zero
 
 local attribute [semireducible] with_zero
 
+/--  If `α` is a Type with `≤`, then `with_zero α` is also a Type with `≤` and `0 ∈ with_zero α`
+is the smallest element. -/
+def has_le [has_le α] : has_le (with_zero α) :=
+{ le := λ o₁ o₂ : option α, ∀ a ∈ o₁, ∃ b ∈ o₂, a ≤ b, }
```

Comment:
Why is `with_zero.preorder` not sufficient? I assume this is definitionally equal to the instances provided by that one?

Reply:
I really do not know: I copied this definition from `order/bounded_lattice` (l. 451), since I assumed that this was standard.

Also, I do not understand how to use `with_zero.preorder`...

Reply:
> Also, I do not understand how to use with_zero.preorder...

Just assume that `α` is a preorder, and then `with_zero α` will be a preorder automatically using that instance. You don't have to do anything special.

Reply:
You can probably just remove this. It would only be useful if you want an order on `with_zero α` when `α` has an order that is not a `preorder` (which seems unlikely).

Reply:
You are right: removing the definition still compiles!